lib/auth/delegation/delegationv1/profile_service.go

```diff
@@ -0,0 +1,223 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+ package delegationv1
+
+import (
+	"context"
+	"log/slog"
+
+	"github.com/gravitational/trace"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	delegationv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/delegation/v1"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/authz"
+)
+
+// ProfileServiceConfig contains the configuration of the ProfileService.
+type ProfileServiceConfig struct {
+	// Authorizer is used to authorize the user.
+	Authorizer authz.Authorizer
+
+	// Writer is used to create, update, and delete profile resources.
+	Writer ProfileWriter
+
+	// Reader is used to read and list profile resources.
+	Reader ProfileReader
```

Comment:
I'd perhaps make the GoDoc on `Reader` a little more explicit that we /expect/ this to usually be the cache.